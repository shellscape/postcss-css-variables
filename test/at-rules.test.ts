import { fixture, process } from './helpers';

const tests = {
  'add rule declaration of property in @support': 'support-directive',
  'work with at-rules containing properties': 'at-rules-containing-properties'
};

describe('at-rules', async () => {
  test.each(Object.entries(tests))('%s', async (_desc, file) => {
    const css = await fixture(`at-rules/${file}`);
    const result = await process(css);
    expect(result).toMatchSnapshot();
  });
});
