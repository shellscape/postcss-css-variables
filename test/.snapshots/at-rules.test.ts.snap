// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`at-rules > add rule declaration of property in @support 1`] = `".box{width:100px}@supports (display: flex){.box{width:200px}}"`;

exports[`at-rules > work with at-rules containing properties 1`] = `"@font-face{font-family:'my-font-family-name';src:url('myfont.woff2') format('woff2')}"`;
